src/dotnet/ToolPackage/ToolPackageInstaller.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.Tools;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class ToolPackageInstaller : IToolPackageInstaller
+    {
+        private const string StagingDirectory = ".stage";
+
+        private readonly IProjectRestorer _projectRestorer;
+        private readonly FilePath? _tempProject;
+        private readonly DirectoryPath _offlineFeed;
+
+        public ToolPackageInstaller(
+            IToolPackageRepository repository,
+            IProjectRestorer projectRestorer,
+            FilePath? tempProject = null,
+            DirectoryPath? offlineFeed = null)
+        {
+            Repository = repository ?? throw new ArgumentNullException(nameof(repository));
+            _projectRestorer = projectRestorer ?? throw new ArgumentNullException(nameof(projectRestorer));
+            _tempProject = tempProject;
+            _offlineFeed = offlineFeed ?? new DirectoryPath(new CliFolderPathCalculator().CliFallbackFolderPath);
+        }
+
+        public IToolPackageRepository Repository { get; private set; }
```

Comment:
Need public getter?

Reply:
I added this property so that I didn't have to pass around the repository (renaming to "store") associated with the installer, primarily for tests that want to immediately query to see if the package installed into the store correctly.

I'd be happy to remove it from the interface if you think it'd be a cleaner design.

Reply:
in the test you can keep the reference of store after the construction of Installer. And use it to assert. Although assert in a lower abstraction is also bad.

But to be practical, let's see if it is hard to remove it. It is ok it keep, since it can only query stuff.

Reply:
Shouldn't be that hard.  I'll remove.

Reply:
Removed in upcoming push.